synapse/rest/admin/rooms.py

```diff
@@ -694,3 +686,55 @@ async def on_GET(
         )
 
         return 200, results
+
+
+async def _delete_room(
+    request: SynapseRequest,
+    room_id: str,
+    auth,
+    room_shutdown_handler,
+    pagination_handler,
```

Comment:
We may as well fully type this function (you'll also need to import things appropriately):

```suggestion
    auth: Auth,
    room_shutdown_handler: RoomShutdownHandler,
    pagination_handler: PaginationHandler,
```

FTR you can see what these types are by looking in `synapse/server.py`